mkinitcpio

```diff
@@ -443,9 +453,25 @@ process_preset() (
             done
         fi
 
-        preset_cmd+=($OPTREST)
-        msg2 "${preset_cmd[*]}"
-        MKINITCPIO_PROCESS_PRESET=1 "$0" "${preset_cmd[@]}"
+        if (( _optremove )); then
+            if (( ${#preset_remove_cmd[*]} )); then
+                # we won't be calling mkinitcpio recursively, so no need to set MKINITCPIO_PROCESS_PRESET
+                for pr in "${preset_remove_cmd[@]}"; do
+                    if [[ ! -f "${pr}" ]]; then
+                        warning "Image not found: \`%s'" "$pr"
+                    elif [[ ! -w "${pr}" ]]; then
+                        error "Image not writable: \`%s'" "$pr"
+                    else
+                        rm -f -- "${pr}"
+                        msg2 "Removed: \`%s'" "$pr"
+                    fi
+                done
+            fi
+        else
+            preset_cmd+=($OPTREST)
+            msg2 "${preset_cmd[*]}"
+            MKINITCPIO_PROCESS_PRESET=1 "$0" "${preset_cmd[@]}"
+        fi
```

Comment:
Currently `mkinitcpio -P linux -- --uefi "lol.efi"` is going to produce an invocation of `mkinitcpio` with all argument after `--` appended.

This isn't semantically correct and not what people expect should happen.

What I'm wondering is if we should instead record all the flags we are parsing and if we see `-p/-P` we should instead append all of these flags on the preset processing. It would allow us to implement `--remove` as a simple `does the outputfile exist? rm and exit 0`.

I suspect this is larger change, and we should instead accept this for the time being then see if we can rework `mkinitcpio` with the above mentioned change later.

Thoughts @nl6720 ?

Reply:
Then we are on the same page :)